### PR TITLE
docs: add separate linkable browser-window-options.md and web-preferences.md

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
`BrowserWindowOptions` and `WebPreferences` are used in multiple places.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added

#### Release Notes
Notes: none

### CODE REVIEW

File: docs/api/window-open.md [hunk 1]
```diff
@@ -33,12 +33,12 @@ because it is invoked in the main process.
...
 
 `features` is a comma-separated key-value list, following the standard format of
-the browser. Electron will parse `BrowserWindowConstructorOptions` out of this
+the browser. Electron will parse [BrowserWindowConstructorOptions](structures/browser-window-options.md) out of this
 list where possible, for convenience. For full control and better ergonomics,
 consider using `webContents.setWindowOpenHandler` to customize the
 BrowserWindow creation.
 
-A subset of `WebPreferences` can be set directly,
+A subset of [WebPreferences](structures/web-preferences.md) can be set directly,
```
Comment (dsanders11) [inline docs/api/window-open.md:41]:
```suggestion
A subset of [`WebPreferences`](structures/web-preferences.md) can be set directly,
```

nit: I think this should keep the backticks.

File: docs/api/window-open.md [hunk 2]
```diff
@@ -33,12 +33,12 @@ because it is invoked in the main process.
 Returns [`Window`](https://developer.mozilla.org/en-US/docs/Web/API/Window) | null
 
 `features` is a comma-separated key-value list, following the standard format of
-the browser. Electron will parse `BrowserWindowConstructorOptions` out of this
+the browser. Electron will parse [BrowserWindowConstructorOptions](structures/browser-window-options.md) out of this
```
Comment (dsanders11) [inline docs/api/window-open.md:36]:
```suggestion
the browser. Electron will parse [`BrowserWindowConstructorOptions`](structures/browser-window-options.md) out of this
```

nit: I think this should keep the backticks.

File: docs/api/window-open.md [hunk 3]
```diff
@@ -60,15 +60,15 @@ window.open('https://github.com', '_blank', 'top=500,left=200,frame=false,nodeIn
   `features` will be passed to any registered `webContents`'s
   `did-create-window` event handler in the `options` argument.
 * `frameName` follows the specification of `windowName` located in the [native documentation](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#parameters).
-* When opening `about:blank`, the child window's `WebPreferences` will be copied
+* When opening `about:blank`, the child window's [WebPreferences](structures/web-preferences.md) will be copied
```
Comment (dsanders11) [inline docs/api/window-open.md:63]:
```suggestion
* When opening `about:blank`, the child window's [`WebPreferences`](structures/web-preferences.md) will be copied
```

nit: I think this should keep the backticks.

File: docs/api/window-open.md [hunk 4]
```diff
@@ -60,15 +60,15 @@ window.open('https://github.com', '_blank', 'top=500,left=200,frame=false,nodeIn
...
   from the parent window, and there is no way to override it because Chromium
   skips browser side navigation in this case.
 
 To customize or cancel the creation of the window, you can optionally set an
 override handler with `webContents.setWindowOpenHandler()` from the main
 process. Returning `{ action: 'deny' }` cancels the window. Returning `{
 action: 'allow', overrideBrowserWindowOptions: { ... } }` will allow opening
-the window and setting the `BrowserWindowConstructorOptions` to be used when
+the window and setting the [BrowserWindowConstructorOptions](structures/browser-window-options.md) to be used when
```
Comment (dsanders11) [inline docs/api/window-open.md:71]:
```suggestion
the window and setting the [`BrowserWindowConstructorOptions`](structures/browser-window-options.md) to be used when
```

nit: I think this should keep the backticks.

File: docs/api/app.md [hunk 5]
```diff
@@ -1353,7 +1353,7 @@ Start accessing a security scoped resource. With this method Electron applicatio
 
 ### `app.enableSandbox()`
 
-Enables full sandbox mode on the app. This means that all renderers will be launched sandboxed, regardless of the value of the `sandbox` flag in WebPreferences.
+Enables full sandbox mode on the app. This means that all renderers will be launched sandboxed, regardless of the value of the `sandbox` flag in [WebPreferences](structures/web-preferences.md).
```
Comment (dsanders11) [inline docs/api/app.md:1356]:
```suggestion
Enables full sandbox mode on the app. This means that all renderers will be launched sandboxed, regardless of the value of the `sandbox` flag in [`WebPreferences`](structures/web-preferences.md).
```

nit: Let's add backticks here while we're at it.

File: docs/api/structures/browser-window-options.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+  the preferred width of the web page when zoomed, `false` will cause it to
+  zoom to the width of the screen. This will also affect the behavior when
+  calling `maximize()` directly. Default is `false`.
+* `tabbingIdentifier` string (optional) _macOS_ - Tab group name, allows
+  opening the window as a native tab on macOS 10.12+. Windows with the same
+  tabbing identifier will be grouped together. This also adds a native new
+  tab button to your window's tab bar and allows your `app` and window to
+  receive the `new-window-for-tab` event.
+* `webPreferences` [WebPreferences](web-preferences.md) (optional) - Settings of web page's features.
+* `titleBarOverlay` Object | Boolean (optional) -  When using a frameless window in conjunction with `win.setWindowButtonVisibility(true)` on macOS or using a `titleBarStyle` so that the standard window controls ("traffic lights" on macOS) are visible, this property enables the Window Controls Overlay [JavaScript APIs][overlay-javascript-apis] and [CSS Environment Variables][overlay-css-env-vars]. Specifying `true` will result in an overlay with default system colors. Default is `false`.
```
Comment (dsanders11) [inline docs/api/structures/browser-window-options.md:123]: There are some broken links in here, like `[JavaScript APIs][overlay-javascript-apis]`.

**Review by dsanders11 — DISMISSED** [review]
I'm also mostly 👍 on this as I've considered the same change before when in the docs, so that's several of us now that have been inclined to make this change.

But I do understand the concerns about the docs being a bit more cumbersome as a result. 🤔 For context, I believe [these are the past comments](https://github.com/electron/electron/pull/23054#discussion_r406388437) that @zcbenz is referencing.

I think that's less of a concern on the current docs website (which wasn't the case when those comments were originally made) since now when you hover over a linked structure, it shows you the details. Although that won't alleviate the concern about `WebPreferences` being not immediately visible on the `BrowserWindow` page since I don't think you can hover in the hove. I think the current docs implementation also gives us more freedom to potentially inline the structure as well if we wanted, so I think there are options.

I'm throwing a "request changes" on this just because there are a handful of broken reference-style links as a result of the changes. Unfortunately the docs linker doesn't catch that they're broken.

**Comment by miniak** [comment]
@dsanders11 should I rebase this one? Does it have a chance being merged?

**Comment by erickzhao** [comment]
I think we want to make some docs-related changes to make this ultimately more readable on the website cc @nornagon

**Comment by dsanders11** [comment]
> Does it have a chance being merged?

But, to expand on @erickzhao's comment, I think the answer to this is yes. Consensus seemed to be reached on we're happy to merge this if we can tweak the website to better handle the UX in the published docs.

**Comment by miniak** [comment]
rebased and linting issues fixed

**Review by codebytere — APPROVED** [review]

**Review by dsanders11 — CHANGES_REQUESTED** [review]
Putting a change on this to prevent merge only because we want to make a change to how it's displayed on the website for UX:

* Ability to inline the structure, for example in `docs/api/browser-window.md` so that users don't have to click through
* Ability to disable the hover card on a structure on the website, so that a very large structure like this one doesn't cause problems

Current plan is to extend the website code to handle special URL queries for this behavior, e.g. `?inline=true` or `?hover=false`,  and I'll try to get a PR up on `electron/website` for that by EOW, or I'll clear this request changes.

Thanks for the patience on this one @miniak.

**Review by dsanders11 — COMMENTED** [review]
Website changes have landed, we should be good to go now. @miniak, could you grab the changes in 5522d5e124afa1ffe402fe8cc8b2f59ff497209c and add them to this PR so that inlining will happen for the website? Once that's done we're good to merge. 👍

**Comment by miniak** [comment]
@dsanders11 done

**Review by dsanders11 — APPROVED** [review]

**Pull request merged**